src/dg/dg.cpp

```diff
@@ -3330,6 +3330,839 @@ real2 DGBase<dim,real,MeshType>::discontinuity_sensor(
+            eps = std::min({ all_parameters->pos_eps, rho_avg, p_avg });
+            if (eps < 0) eps = all_parameters->pos_eps;
+        }
+
+        real theta = 1.0;
+        if (soln_cell_avg[0] - local_min[0] > 1e-13)
+            theta = (soln_cell_avg[0] - local_min[0] == 0) ? 1.0 : std::min( (soln_cell_avg[0] - eps) / (soln_cell_avg[0] - local_min[0]), 1.0);
+
+        //apply limiter on density values at quadrature points
+        for (unsigned int istate = 0; istate < 1/*nstate*/; ++istate) {
+            for (unsigned int iquad = 0; iquad < n_quad_pts; ++iquad) {
+                if (isnan(soln_at_q[istate][iquad]))
+                {
+                    std::cout << " Density is NaN - Aborting... " << std::endl << std::flush;
+                    std::abort();
+                }
+
+                soln_at_q[istate][iquad] = theta * (soln_at_q[istate][iquad] - soln_cell_avg[istate])
+                    + soln_cell_avg[istate];
+            }
+        }
+
+        if (nstate == PHILIP_DIM + 2)
+        {
+             std::vector< real > p_lim(n_quad_pts);
+             for (unsigned int iquad = 0; iquad < n_quad_pts; ++iquad) {
+                p_lim[iquad] = (gamma-1) * (soln_at_q[nstate - 1][iquad] - 
+                                0.5 * ((soln_at_q[1][iquad] * soln_at_q[1][iquad]) / soln_at_q[0][iquad]));
+                if (dim > 1)
+                    p_lim[iquad] += (gamma-1)*(- 0.5 * ((soln_at_q[2][iquad] * soln_at_q[2][iquad]) / soln_at_q[0][iquad]));
+                if (dim > 2)
+                    p_lim[iquad] += (gamma-1)*(- 0.5 * ((soln_at_q[3][iquad] * soln_at_q[3][iquad]) / soln_at_q[0][iquad]));
+             }
+
+            std::vector<real> theta2(n_quad_pts, 1);
+            for (unsigned int iquad = 0; iquad < n_quad_pts; ++iquad) {
+                if (p_lim[iquad] >= eps) 
+                {
+                    theta2[iquad] = 1;
+                }
+                else 
+                {
+                    real a = (soln_at_q[nstate - 1][iquad] - soln_cell_avg[nstate - 1]) * (soln_at_q[0][iquad] - soln_cell_avg[0]) - 0.5 * pow(soln_at_q[1][iquad] - soln_cell_avg[1], 2);
+
+                    real b = soln_cell_avg[0] * (soln_at_q[nstate - 1][iquad] - soln_cell_avg[nstate - 1]) + soln_cell_avg[nstate - 1] * (soln_at_q[0][iquad] - soln_cell_avg[0])
+                        - soln_cell_avg[1] * (soln_at_q[1][iquad] - soln_cell_avg[1]) - (eps / gamma) * (soln_at_q[0][iquad] - soln_cell_avg[0]);
+
+                    real c = (soln_cell_avg[nstate - 1] * soln_cell_avg[0]) - 0.5 *pow(soln_cell_avg[1],2) - (eps / gamma) * soln_cell_avg[0];
+
+                    if (dim > 1)
+                    {
+                        a -= 0.5 * pow(soln_at_q[2][iquad] - soln_cell_avg[2], 2);
+                        b -= soln_cell_avg[2] * (soln_at_q[2][iquad] - soln_cell_avg[2]);
+                        c -= 0.5 * pow(soln_cell_avg[2], 2);
+                    }
```

Comment:
Avoid naming variables like `a`, `b`, `c`. If they are taken directly out of a reference, please link the reference the first time those variables are defined.

Reply:
Changed the naming of those variables to s_coeff1,s_coeff2,etc. Only kept one variable named `a` within the TVB limiter as that is what is used in the reference. When the variable is first defined, the reference is listed as the name of the authors, year and equation number. Will this suffice or is a link to the reference required?